### ISSUE TITLE
[question]: pvc resizing cannot currently modify

### ISSUE THREAD

**Issue by wesbragagt** [Panfactum/stack#174]
### Prior Search

- [X] I have already searched this [project's issues](https://github.com/Panfactum/stack/issues) to determine if a similar question has already been asked.

### What is your question?

Our prouduction database is stuck creating a replica due to a pvc resize event failing:

I'm unsure how to move forward.

```yaml
# persistentvolume claim yaml spec
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    cnpg.io/nodeSerial: "3"
    cnpg.io/operatorVersion: 1.23.1
    cnpg.io/pvcStatus: ready
    config.linkerd.io/skip-inbound-ports: "5432"
    linkerd.io/inject: enabled
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    resize.topolvm.io/increase: 10Gi
    resize.topolvm.io/pre_capacity_bytes: "115658252288"
    resize.topolvm.io/storage_limit: 1000Gi
    resize.topolvm.io/threshold: 35%
    volume.beta.kubernetes.io/storage-provisioner: ebs.csi.aws.com
    volume.kubernetes.io/selected-node: ip-10-0-156-66.us-west-2.compute.internal
    volume.kubernetes.io/storage-provisioner: ebs.csi.aws.com
  creationTimestamp: "2024-11-21T17:51:29Z"
  finalizers:
  - kubernetes.io/pvc-protection
  - snapshot.storage.kubernetes.io/pvc-as-source-protection
  labels:
    cnpg.io/cluster: pg-88cc
    cnpg.io/instanceName: pg-88cc-3
    cnpg.io/instanceRole: replica
    cnpg.io/pvcRole: PG_DATA
    id: pg-pg-88cc-295bc33c00cc195b
    panfactum.com/prevent-lifetime-eviction: "true"
    panfactum.com/scheduler: "true"
    panfactum.com/workload: pg-pg-88cc
    pg-cluster: implentio-pg-88cc
    role: replica
  name: pg-88cc-3
  namespace: implentio
  ownerReferences:
  - apiVersion: postgresql.cnpg.io/v1
    controller: true
    kind: Cluster
    name: pg-88cc
    uid: 7191bf16-0c11-42b6-9852-746d96374a4b
  resourceVersion: "182060977"
  uid: 3cf283e2-ced1-46e6-913b-9e12a4f3eafe
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 120Gi
  storageClassName: ebs-standard
  volumeMode: Filesystem
  volumeName: pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 110Gi
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2024-11-21T18:27:25Z"
    status: "True"
    type: Resizing
  phase: Bound

```

```yaml
# persistentvolumeclaim describe
Name:          pg-88cc-3
Namespace:     implentio
StorageClass:  ebs-standard
Status:        Bound
Volume:        pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe
Labels:        cnpg.io/cluster=pg-88cc
               cnpg.io/instanceName=pg-88cc-3
               cnpg.io/instanceRole=replica
               cnpg.io/pvcRole=PG_DATA
               id=pg-pg-88cc-295bc33c00cc195b
               panfactum.com/prevent-lifetime-eviction=true
               panfactum.com/scheduler=true
               panfactum.com/workload=pg-pg-88cc
               pg-cluster=implentio-pg-88cc
               role=replica
Annotations:   cnpg.io/nodeSerial: 3
               cnpg.io/operatorVersion: 1.23.1
               cnpg.io/pvcStatus: ready
               config.linkerd.io/skip-inbound-ports: 5432
               linkerd.io/inject: enabled
               pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               resize.topolvm.io/increase: 10Gi
               resize.topolvm.io/pre_capacity_bytes: 115658252288
               resize.topolvm.io/storage_limit: 1000Gi
               resize.topolvm.io/threshold: 35%
               volume.beta.kubernetes.io/storage-provisioner: ebs.csi.aws.com
               volume.kubernetes.io/selected-node: ip-10-0-156-66.us-west-2.compute.internal
               volume.kubernetes.io/storage-provisioner: ebs.csi.aws.com
Finalizers:    [kubernetes.io/pvc-protection snapshot.storage.kubernetes.io/pvc-as-source-protection]
Capacity:      110Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Used By:       pg-88cc-3-join-w9dvc
Conditions:
  Type       Status  LastProbeTime                     LastTransitionTime                Reason  Message
  ----       ------  -----------------                 ------------------                ------  -------
  Resizing   True    Mon, 01 Jan 0001 00:00:00 +0000   Thu, 21 Nov 2024 12:17:25 -0600           
Events:
  Type     Reason                      Age                 From                                                                                     Message
  ----     ------                      ----                ----                                                                                     -------
  Normal   WaitForPodScheduled         35m (x6 over 35m)   persistentvolume-controller                                                              waiting for pod pg-88cc-3-join-w9dvc to be scheduled
  Warning  ProvisioningFailed          34m (x5 over 34m)   ebs.csi.aws.com_ebs-csi-controller-8f587579f-4nk6n_4a93038c-da77-45ba-8fc5-1a41d7b70db6  failed to provision volume with StorageClass "ebs-standard": error generating accessibility requirements: no topology key found on CSINode ip-10-0-156-66.us-west-2.compute.internal
  Normal   ExternalProvisioning        34m (x3 over 34m)   persistentvolume-controller                                                              Waiting for a volume to be created either by the external provisioner 'ebs.csi.aws.com' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered.
  Normal   Provisioning                34m (x6 over 34m)   ebs.csi.aws.com_ebs-csi-controller-8f587579f-4nk6n_4a93038c-da77-45ba-8fc5-1a41d7b70db6  External provisioner is provisioning volume for claim "implentio/pg-88cc-3"
  Normal   ProvisioningSucceeded       34m                 ebs.csi.aws.com_ebs-csi-controller-8f587579f-4nk6n_4a93038c-da77-45ba-8fc5-1a41d7b70db6  Successfully provisioned volume pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe
  Normal   Resized                     24m                 pvc-autoresizer                                                                          PVC volume is resized to 110Gi
  Normal   FileSystemResizeRequired    24m                 external-resizer ebs.csi.aws.com                                                         Require file system resize of volume on node
  Normal   FileSystemResizeSuccessful  23m                 kubelet                                                                                  MountVolume.NodeExpandVolume succeeded for volume "pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe" ip-10-0-156-66.us-west-2.compute.internal
  Warning  ExternalExpanding           23m (x2 over 24m)   volume_expand                                                                            waiting for an external controller to expand this PVC
  Normal   Resized                     23m                 pvc-autoresizer                                                                          PVC volume is resized to 120Gi
  Normal   Resizing                    10m (x14 over 24m)  external-resizer ebs.csi.aws.com                                                         External resizer is resizing volume pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe
  Warning  VolumeResizeFailed          10m (x12 over 23m)  external-resizer ebs.csi.aws.com                                                         resize volume "pvc-3cf283e2-ced1-46e6-913b-9e12a4f3eafe" by resizer "ebs.csi.aws.com" failed: rpc error: code = Internal desc = Could not resize volume "vol-0d1843ab186ea47a7": rpc error: code = Internal desc = Could not modify volume "vol-0d1843ab186ea47a7": volume "vol-0d1843ab186ea47a7" in OPTIMIZING state, cannot currently modify
```

### What primary components of the stack does this relate to?

kube_pg_cluster

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wesbragagt** [comment]
I'm also noticing various ebs volumes that are sitting available. I'm wondering if this could be causing a failure on the pvc resizer. Should I delete the ones that are not attached to a node? 
![image](https://github.com/user-attachments/assets/611c9059-6e61-4e57-ba04-01507d5deb1d)

**Comment by wesbragagt** [comment]
I've scaled down my database to 1, deleted pv and pvcs in k9s and attempted an apply to increase back to 2 and now I'm getting in my pvc-controller:

```pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.webhook","msg":"Registering webhook","path":"/pvc/mutate"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"setup","msg":"starting manager"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.metrics","msg":"Starting metrics server"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.metrics","msg":"Serving metrics server","bindAddress":":8080","secure":false}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.webhook","msg":"Starting webhook server"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","msg":"starting server","kind":"health probe","addr":":8081"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.webhook","msg":"Serving webhook server","host":"","port":9443}
Stream closed EOF for pvc-autoresizer/pvc-autoresizer-controller-66f7485785-752cj (linkerd-init)
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:09Z","logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
pvc-autoresizer I1121 19:40:09.359658       1 leaderelection.go:250] attempting to acquire leader lease pvc-autoresizer/49e22f61.topolvm.io...
pvc-autoresizer I1121 19:40:26.780981       1 leaderelection.go:260] successfully acquired lease pvc-autoresizer/49e22f61.topolvm.io
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:37Z","logger":"pvc-autoresizer.resize","msg":"resize started","namespace":"implentio","name":"pg-88cc-4","from":107374182400,"to":118111600640,"threshold":36781241548,"available":33716604928,"inodesThreshold":655360,"inodesAvailable":6552142}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:47Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":105089261568}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:40:56Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":105089261568}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:06Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":105089261568}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:16Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":105089261568}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:27Z","logger":"pvc-autoresizer.resize","msg":"resize started","namespace":"implentio","name":"pg-88cc-4","from":118111600640,"to":128849018880,"threshold":40480388300,"available":34937470976,"inodesThreshold":720896,"inodesAvailable":7207387}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:37Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:47Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:41:57Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:06Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:16Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:27Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:36Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:47Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:42:56Z","logger":"pvc-autoresizer.resize","msg":"waiting for resizing...","namespace":"implentio","name":"pg-88cc-4","capacity":115658252288}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:07Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:16Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:27Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:36Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:47Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:43:56Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
pvc-autoresizer {"level":"info","ts":"2024-11-21T19:44:07Z","logger":"pvc-autoresizer","msg":"failed to get volume stats","namespace":"implentio","name":"pg-88cc-4"}
```

**Comment by fullykubed** [comment]
AWS has a limitation that prevents resizing more than once every 6 hours.

See the docs here: https://panfactum.com/docs/edge/reference/infrastructure-modules/submodule/kubernetes/kube_pg_cluster#storage
